modules/server/src/test/scala/scaladex/server/util/PaginatedResponseReader.scala

```diff
@@ -0,0 +1,14 @@
+package scaladex.server.util
+
+import play.api.libs.json.Json
+import play.api.libs.json.Reads
+import scaladex.core.model.search.Page
+import scaladex.core.model.search.Pagination
+
+trait PaginatedResponseReader {
+
+  implicit def pageItemsJson[A](implicit items: Reads[A]): Reads[Seq[A]] = Reads.seq(items)
+  implicit val paginationJson: Reads[Pagination] = Json.reads[Pagination]
+  implicit def pageJson[A](implicit items: Reads[A]): Reads[Page[A]] = Json.reads[Page[A]]
```

Comment:
The unit tests are failing before they're even running because of [this error](https://github.com/scalacenter/scaladex/runs/5958735189?check_suite_focus=true#step:5:503), which reports `items` here as being unused.

I think this is needed, otherwise `pageJson` fails to find an implicit value for `A`. 

I can mark this with a compiler annotation `@nowarn`, but I'd be interested to see if there's a way around this, maybe I'm not understanding implicits here correctly.

Reply:
I recommend not to generate JSON codecs other than the ones you can get by interpreting the schemas defined in `PaginationSchema`. This ensures that the codecs you use for testing are the same as the codecs used in the app.

Reply:
Thanks for the response. Could you clarify what you mean by 

>  interpreting the schemas defined in `PaginationSchema`

There's only a single schema defined there, which is `Pagination`. Wouldn't we still have to define a codec for the `Page` case class as well? Sorry for all the questions, I appreciate your help :)

Reply:
Hey, you are totally right, we also need to define a schema for `Page` :) Currently, how are pages serialized in HTTP responses?

Interpreting a schema means mixing an implementation trait into a trait that contains schema definitions:

~~~ scala
object PlayJsonSchemas extends PaginationSchema with endpoints4s.playjson.JsonSchemas

// And then
PlayJsonSchemas.paginationSchema.reads // : Reads[Pagination]
~~~

I just realized that the trait `PaginationSchema` extends `JsonEntitiesFromSchemas`, but it should only extend `JsonSchemas` to be able to apply the play-json interpreter as shown above.

Let me know if you have other questions.

Reply:
Currently, `Page` is serialized like this in `ArtifactEndpoints`

```scala
  implicit val artifactEndpointResponse: JsonSchema[Page[ArtifactResponse]] =
    field[Pagination]("pagination")
      .zip(field[Seq[ArtifactResponse]]("items"))
      .xmap[Page[ArtifactResponse]] { case (pagination, artifacts) => Page(pagination, artifacts) } {
        case Page(pagination, artifacts) => (pagination, artifacts)
      }

```

I think I'm getting close with the `PlayJsonSchemas` object, 

```scala
trait PaginatedResponseReader {

  implicit val readPagination: Reads[Pagination] = PlayJsonCodecs.paginationSchema.reads

}
```

But I'm unsure of how to interpret a `Page[ArtifactResponse]`, since it's a generic case class. There's also the problem of [this line of code](https://github.com/scalacenter/scaladex/pull/992/files#diff-5cf5e83486098319404a7d13272635d19c70ffe281deaad2b5395c45e32a4cf9R38), which no longer compiles given that `PaginationSchema` no longer extends `JsonEntitiesFromSchemas`.

I guess `ArtifactEndpoints` itself could extend `JsonEntitiesFromSchemas`, but I think that leads to the problem of generate JSON codecs other than the ones you can get by interpreting the schemas defined in `PaginationSchema`, which is what we want to avoid in the first place.